The documented CMake FlatBuffers version override is incorrect and doesn't work when corrected

### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

v2.19.1

### Custom code

Yes

### OS platform and distribution

Linux arm64v8/ubuntu:22.04

### Mobile device

_No response_

### Python version

3.10.12

### Bazel version

Bazelisk 1.25.0 → Bazel 6.5.0 

### GCC/compiler version

11.4.0

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?


`OverridableFetchContent_Declare()` documents a way to override a fetched
dependency's `GIT_TAG` (among other args). For `flatbuffers` specifically,
following that documentation does not let you build TFLite against a
different FlatBuffers version, for two independent reasons:

- A: The documented variable name is wrong
  -  the doc's own example contradicts its own stated convention, so following it silently does nothing.
- B: Even using the correct variable name, the override doesn't actually work for FlatBuffers
  - TFLite fails to compile against any version  other than the one it currently vendors, because of a checked-in,  hardcoded version assertion.

Filing both together because a reader who only fixes (A) will reasonably conclude the override now works 
(CMake stops warning about an unused variable) 
and only discover (B) later, at a compile failure that gives no indication it's related to the override at all.

### A: Wrong variable name in the doc comment

`tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake`'s doc
comment for `OverridableFetchContent_Declare()` states the override
convention is `OVERRIDABLE_FETCH_CONTENT_<contentName>_<variable>`, but then
gives a contradictory example that omits the prefix:

```cmake
# It's possible to override, GIT_REPOSITORY, GIT_TAG, URL and URL_HASH for
# a target by setting
# OVERRIDABLE_FETCH_CONTENT_<contentName>_<variable> where <contentName> is the
# CONTENT_NAME argument content provided to this method and <variable> is the
# argument of this method to override. For example, given CONTENT_NAME = foo
# the GIT_REPOSITORY can be overridden by setting foo_GIT_REPOSITORY to the
# value to use instead.
```

The actual implementation only reads the prefixed form:

```cmake
foreach(OVERRIDABLE_ARG in ${OVERRIDABLE_ARGS})
  set(OVERRIDE_VALUE
    ${OVERRIDABLE_FETCH_CONTENT_${CONTENT_NAME}_${OVERRIDABLE_ARG}}
  )
  ...
```

Following the doc's own example — e.g. `-Dflatbuffers_GIT_TAG=<tag>` to
override the `flatbuffers` content's `GIT_TAG` when configuring
`tensorflow/lite` standalone via CMake — silently does nothing. CMake reports:

```
CMake Warning:
  Manually-specified variables were not used by the project:
    flatbuffers_GIT_TAG
```

and the dependency is fetched at its originally pinned commit/tag regardless.
The variable that actually works — i.e. that CMake accepts without warning
and that changes which ref gets fetched — is
`OVERRIDABLE_FETCH_CONTENT_flatbuffers_GIT_TAG`.

**Repro:**

```bash
git clone -b v2.19.1 --depth 1 https://github.com/tensorflow/tensorflow.git
cmake -B tflite-build -S tensorflow/tensorflow/lite \
    -DCMAKE_BUILD_TYPE=Release \
    -Dflatbuffers_GIT_TAG=v24.3.7   # per the doc's own example — silently ignored
```
vs.
```bash
cmake -B tflite-build -S tensorflow/tensorflow/lite \
    -DCMAKE_BUILD_TYPE=Release \
    -DOVERRIDABLE_FETCH_CONTENT_flatbuffers_GIT_TAG=v24.3.7   # accepted, no warning
```

**Expected behavior:** the doc comment's example should use the correct
prefixed variable name, or the code should also accept the unprefixed form
the doc currently describes (for anyone who already followed it).

### B: Even with the correct variable, overriding FlatBuffers doesn't work

Once A is fixed and the override is actually applied — CMake fetches the requested `GIT_TAG` and no longer warns. The build still fails, because the checked-in generated header `tensorflow/compiler/mlir/lite/schema/schema_generated.h` hard-asserts an *exact* FlatBuffers version match:

```cpp
static_assert(FLATBUFFERS_VERSION_MAJOR == 25 &&
              FLATBUFFERS_VERSION_MINOR == 9 &&
              FLATBUFFERS_VERSION_REVISION == 23,
              "Non-compatible flatbuffers version included");
```

```
error: static assertion failed: Non-compatible flatbuffers version included
```

So, in practice, `OverridableFetchContent`'s FlatBuffers override capability
is non-functional for its apparent purpose (building TFLite against a
different, org-standard FlatBuffers release) regardless of which variable
name is used — the only value that will ever compile is whatever exact
version the checked-in schema header happens to have been generated with.

This is the same root cause already reported in #62298 ("schema_generated.h
checked into source instead of generated"), closed `not_planned`. I'm not
expecting (B) to be separately actionable given that prior decision — including
it here so the doc/variable-name fix in (A) doesn't get made in a way that
implies the override is actually usable for FlatBuffers, when it isn't.

### Suggested resolution

At minimum, fix A so the documented example matches the implementation.
Given B, consider also documenting on `flatbuffers.cmake` itself (not just
the generic `OverridableFetchContent.cmake` helper) that overriding
`flatbuffers_GIT_TAG`/`OVERRIDABLE_FETCH_CONTENT_flatbuffers_GIT_TAG` is
expected to fail to compile for any version other than the one currently
vendored, given #62298's resolution — so the next person doesn't spend time
rediscovering (B) after fixing (A).

### Environment

- Repro'd against TensorFlow tag `v2.19.1`.
- Confirmed both A and B are still present as of `master` (checked  2026-07-31), where the required exact version is now `v25.9.23`.


### Standalone code to reproduce the issue

```shell
Attempt 1:

cmake -B tflite-build -S tensorflow/tensorflow/lite \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
    -DTFLITE_ENABLE_GPU=OFF \
    -DTFLITE_ENABLE_NNAPI=ON \
    -DTFLITE_ENABLE_XNNPACK=ON \
    -Dflatbuffers_GIT_TAG=v24.3.7


and then corrected attempt 2:


cmake -B tflite-build -S tensorflow/tensorflow/lite \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
    -DTFLITE_ENABLE_GPU=OFF \
    -DTFLITE_ENABLE_NNAPI=ON \
    -DTFLITE_ENABLE_XNNPACK=ON \
    -DOVERRIDABLE_FETCH_CONTENT_flatbuffers_GIT_TAG=v24.3.7
```

### Relevant log output

```shell
# Attempt 1: -Dflatbuffers_GIT_TAG=v24.3.7 (wrong/unprefixed name)
# Pipeline 21208, job 108992 — build SUCCEEDED but flag was silently ignored
############################################

No microkernel found in src/reference/packing.cc
No microkernel found in src/reference/unary-elementwise.cc
--
-- 3.21.9.0
-- Performing Test protobuf_HAVE_LD_VERSION_SCRIPT
-- Performing Test protobuf_HAVE_LD_VERSION_SCRIPT - Success
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS - Success
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    flatbuffers_GIT_TAG

...
390:-- Proceeding with version: 24.3.25.4
710:-- Proceeding with version: 24.3.25.4




############################################
# Attempt 2: -DOVERRIDABLE_FETCH_CONTENT_flatbuffers_GIT_TAG=v24.3.7 (correct name)
# Pipeline 21210, job 108999 — CMake accepted it, but the build FAILED to compile
############################################

In file included from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/schema/schema_generated.h:18,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/core/api/op_resolver.h:29,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/core/subgraph.h:36,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc:49:
/builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/compiler/mlir/lite/schema/schema_generated.h:27:44: error: static assertion failed: Non-compatible flatbuffers version included
   27 |               FLATBUFFERS_VERSION_REVISION == 25,
[ 93%] Linking CXX executable protoc

--- full compile error context ---
[ 93%] Building CXX object _deps/abseil-cpp-build/absl/flags/CMakeFiles/absl_flags_reflection.dir/reflection.cc.o
[ 93%] Linking CXX static library libprotoc.a
[ 93%] Built target libprotoc
[ 93%] Building CXX object _deps/protobuf-build/CMakeFiles/protoc.dir/src/google/protobuf/compiler/main.cc.o
In file included from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/schema/schema_generated.h:18,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/core/api/op_resolver.h:29,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/core/subgraph.h:36,
                 from /builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc:49:
/builds/common-external-libs/perception/tflite-build/tensorflow/tensorflow/compiler/mlir/lite/schema/schema_generated.h:27:44: error: static assertion failed: Non-compatible flatbuffers version included
   27 |               FLATBUFFERS_VERSION_REVISION == 25,
[ 93%] Linking CXX executable protoc
[ 93%] Linking CXX static library libabsl_status.a
[ 93%] Built target absl_status
```

--- a/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
+++ b/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
@@ -220,8 +220,8 @@ endfunction()
 # OVERRIDABLE_FETCH_CONTENT_<contentName>_<variable> where <contentName> is the
 # CONTENT_NAME argument content provided to this method and <variable> is the
 # argument of this method to override. For example, given CONTENT_NAME = foo
-# the GIT_REPOSITORY can be overridden by setting foo_GIT_REPOSITORY to the
-# value to use instead.
+# the GIT_REPOSITORY can be overridden by setting
+# OVERRIDABLE_FETCH_CONTENT_foo_GIT_REPOSITORY to the value to use instead.
 #
 # To convert a GIT_REPOSITORY / GIT_TAG reference to a URL,
 # set OVERRIDABLE_FETCH_CONTENT_GIT_REPOSITORY_AND_TAG_TO_URL_<contentName>

--- a/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
+++ b/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
@@ -259,8 +259,8 @@ function(OverridableFetchContent_Declare CONTENT_NAME)
     ${ARGN}
   )
   # Optionally override parsed arguments with values from variables in the form
-  # ${CONTENT_NAME}_${OVERRIDABLE_ARG}.
-  foreach(OVERRIDABLE_ARG in ${OVERRIDABLE_ARGS})
+  # OVERRIDABLE_FETCH_CONTENT_${CONTENT_NAME}_${OVERRIDABLE_ARG}.
+  foreach(OVERRIDABLE_ARG ${OVERRIDABLE_ARGS})
     set(OVERRIDE_VALUE
       ${OVERRIDABLE_FETCH_CONTENT_${CONTENT_NAME}_${OVERRIDABLE_ARG}}
     )

